backend-shared/plannershared.cpp

```diff
@@ -74,16 +74,30 @@ void plannerShared::set_descrate(int value)
 }
 
 // Planning values
-deco_mode plannerShared::planner_deco_mode()
+int plannerShared::planner_deco_mode()
```

Comment:
so this moves the function return value from the `enum` to an `int`
why is that the right thing to do?

Reply:
or, in other words, isn't this a case where this "unification" makes the core code "less typed"?

Reply:
no it does not affect the core code, it makes it available to QML. Deco_mode is not known to 
QML. The duplicated enum in typeenums.h are available but we do not want to use that in desktop-widgets.

This is btw. a good example where having a pure interface class solely for qml makes a lot of sense. Let me rework foundation1 to make plannerShared type secure with the normal enums and at the same time make the functions available to QML.